addons/api/addon/utils/sort-api-records.js

```
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+// created_time is not an attribute defined in models perse, but present in
+// all the resources coming from API. It is the default sorting attribute
+const SORT_DEFAULT_ATTRIBUTE = 'created_time';
+const SORT_DIRECTION_ASCENDING = 'asc';
+const SORT_DIRECTION_DESCENDING = 'desc';
+
+const sortFunctions = {
+  string: (a, b) => String(a).localeCompare(String(b)),
+  date: (a, b) => new Date(a).getTime() - new Date(b).getTime(),
+};
+
+export const sortApiRecords = (records, { querySort, schema }) => {
+  // The consumer is already doing this, not sure is necessary
+  querySort = querySort ?? {};
+
+  // Why the spike uses an assertion to check the attribute exists? - Ask Chad about it.
+  // Perhaps what we should do is throw an error if the attribute does not exists?
```

`assert` from `@ember/debug` does throw an error if it doesn't exist (if that's what the assertion tests).

```js
assert("assertion text to show in error, truthyCaseToAssert);
```

is roughly the same as
```js
if (!truthyCase) {
  throw new Error("error text");
}
```

You get one special ability with `assert` from `@ember/debug` thought. It gets stripped out in production builds. This means in dev and tests you'll get the feedback but in production you won't which is a good thing if you don't know handle the error but also don't want to crash

---

@hashicc 

Very interesting, I didn't though about the assert being stripped out in production code.

So my initial though here was let's throw an error, but your comment got me thinking. If we throw an error, are we going to deal with the error or just throw the error? And if so, do we inform the user this error is happening? are we going to log this error anywhere?

The answers to those questions, right now, is NO. We are not informing the user, neither handling the error. Therefore, the usage of assert will throw the error while testing but not in production.

Unless you think we should be doing something with this kind of errors, I think going with assertion is the way to go. What are your thoughts?